scripts/check_ecosystem.py

```diff
@@ -60,39 +60,50 @@ async def clone(self: Self, checkout_dir: Path) -> AsyncIterator[Path]:
             "--no-tags",
         ]
         if self.ref:
-            git_command.extend(["--branch", self.ref])
+            git_clone_command.extend(["--branch", self.ref])
 
-        git_command.extend(
+        git_clone_command.extend(
             [
                 f"https://github.com/{self.org}/{self.repo}",
                 checkout_dir,
             ],
         )
 
-        process = await create_subprocess_exec(
-            *git_command,
+        git_clone_process = await create_subprocess_exec(
+            *git_clone_command,
             env={"GIT_TERMINAL_PROMPT": "0"},
         )
 
-        status_code = await process.wait()
+        status_code = await git_clone_process.wait()
 
         logger.debug(
             f"Finished cloning {self.org}/{self.repo} with status {status_code}",
         )
+        yield await self._get_commit(checkout_dir)
 
-        yield Path(checkout_dir)
-
-    def url_for(self: Self, path: str, lnum: int | None = None) -> str:
-        """Return the GitHub URL for the given path and line number, if given."""
+    def url_for(self: Self, commit_sha: str, path: str, lnum: int | None = None) -> str:
+        """
+        Return the GitHub URL for the given commit, path, and line number, if given.
+        """
         # Default to main branch
-        url = (
-            f"https://github.com/{self.org}/{self.repo}"
-            f"/blob/{self.ref or 'main'}/{path}"
-        )
+        url = f"https://github.com/{self.org}/{self.repo}/blob/{commit_sha}/{path}"
         if lnum:
             url += f"#L{lnum}"
         return url
 
+    async def _get_commit(self: Self, checkout_dir: Path) -> str:
+        """Return the commit sha for the repository in the checkout directory."""
+        git_sha_process = await create_subprocess_exec(
```

Comment:
you can use `subprocess.check_output` with `text=True` here, it will also check the output status automatically 

https://github.com/astral-sh/ruff/blob/0666added9902990b6ea22d1a48f6fa1ae5ea605/scripts/update_schemastore.py#L18

Reply:
Hm I'd rather not make a blocking call in an async function though 👿

Reply:
fair